salt/netapi/rest_cherrypy/app.py

```diff
@@ -898,6 +899,8 @@ def hypermedia_handler(*args, **kwargs):
     # Transform the output from the handler into the requested output format
     cherrypy.response.headers['Content-Type'] = best
     out = cherrypy.response.processors[best]
+    if isinstance(ret, dict):
```

Comment:
```suggestion
if best == 'application/json' and isinstance(ret, dict):
```
There may be a better way to write that, but the `ct_out_map` could be `application/x-yaml` which should support binary data, and we shouldn't be encoding the bytes there.